src/System.Net.Security/tests/FunctionalTests/ServerAsyncAuthenticateTest.cs

```diff
@@ -0,0 +1,197 @@
+        {
+            yield return new object[] { SslProtocols.Tls, SslProtocols.Tls11, typeof(AuthenticationException)};
+            yield return new object[] { SslProtocols.Tls, SslProtocols.Tls12, typeof(AuthenticationException)};
+            yield return new object[] { SslProtocols.Tls11, SslProtocols.Tls, typeof(TimeoutException) };
+            yield return new object[] { SslProtocols.Tls11, SslProtocols.Tls12, typeof(AuthenticationException) };
+            yield return new object[] { SslProtocols.Tls12, SslProtocols.Tls, typeof(TimeoutException) };
+            yield return new object[] { SslProtocols.Tls12, SslProtocols.Tls11, typeof(TimeoutException) };
+        }
+
+        #region Helpers
+
+        private async Task ServerAsyncSslHelper(
+            SslProtocols clientSslProtocols,
+            SslProtocols serverSslProtocols,
+            bool expectedToFail = false)
+        {
+            _log.WriteLine(
+                "Server: " + serverSslProtocols + "; Client: " + clientSslProtocols + 
+                " expectedToFail: " + expectedToFail);
+
+            int timeOut = expectedToFail ? TestConfiguration.FailingTestTimeoutMiliseconds 
+                : TestConfiguration.PassingTestTimeoutMilliseconds;
+
+            IPEndPoint endPoint = new IPEndPoint(IPAddress.IPv6Loopback, 0);
+            var server = new TcpListener(endPoint);
+            server.Start();
+
+            using (var clientConnection = new TcpClient(AddressFamily.InterNetworkV6))
+            {
+                IPEndPoint serverEndPoint = (IPEndPoint)server.LocalEndpoint;
+
+                Task clientConnect = clientConnection.ConnectAsync(serverEndPoint.Address, serverEndPoint.Port);
+                Task<TcpClient> serverAccept = server.AcceptTcpClientAsync();
+
+                // We expect that the network-level connect will always complete.
+                Task.WaitAll(
+                    new Task[] { clientConnect, serverAccept }, 
+                    TestConfiguration.PassingTestTimeoutMilliseconds);
+
+                using (TcpClient serverConnection = await serverAccept)
+                using (SslStream sslClientStream = new SslStream(clientConnection.GetStream()))
+                using (SslStream sslServerStream = new SslStream(
+                    serverConnection.GetStream(),
+                    false,
+                    AllowAnyServerCertificate))
+                {
+                    string serverName = _serverCertificate.GetNameInfo(X509NameType.SimpleName, false);
+
+                    Task clientAuthentication = sslClientStream.AuthenticateAsClientAsync(
+                        serverName,
+                        null,
+                        clientSslProtocols,
+                        false);
+
+                    Task serverAuthentication = sslServerStream.AuthenticateAsServerAsync(
+                        _serverCertificate,
+                        true,
```

Comment:
This is a good point. I'll investigate why this allows some of the connections.

CI passing is a requirement before merge and all tests were marked as passing.